### PR TITLE
ft(S3C-3002): Add checkpoint creation

### PR DESCRIPTION
**Pull request by tmacro**

### CODE REVIEW

**Review by dora-korpar — APPROVED** [review]

File: warpscript/utapi/decodeCheckpoint.mc2 [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  int64 objD = 2;
+  int64 inB = 3;
+  int64 outB = 4;
+  map<string, int64> ops = 5;
+}
+'>
+PROTOC 'proto' STORE
+<%
+  HEX-> !$proto 'Checkpoint' PB->
+%>
```
Comment (jonathan-gramain) [inline warpscript/utapi/decodeCheckpoint.mc2:14]: Maybe a simple unit test like you did for the snapshot decode/encode could be good to have, same for the other decode/encode macros

File: warpscript/utapi/sumField.mc2 [hunk 2]
```diff
@@ -0,0 +1,46 @@
+{
```
Comment (jonathan-gramain) [inline warpscript/utapi/sumField.mc2]: Shouldn't this go in warpscript/util directory?

File: warpscript/utapi/createCheckpoint.mc2 [hunk 3]
```diff
@@ -0,0 +1,155 @@
+{
```
Comment (jonathan-gramain) [inline warpscript/utapi/createCheckpoint.mc2:1]: This macro looks much like the snapshot one, is there a way to reuse parts of the same code somehow? I'm a noob in mc2 so just take this as a suggestion.

Comment (tmacro) [reply warpscript/utapi/createCheckpoint.mc2:1]: There's definitely some generalization that can be done, but as I'm still a noob as well, I've tried to write it in a straightforward manner to try and salvage what readability I can. When I add the repair process in a later PR, I plan to do some refactoring as the requirements will be a bit more clear.

File: warpscript/utapi/sumField.mc2 (hunk 2 above)
Comment (tmacro) [reply warpscript/utapi/sumField.mc2]: It's already moved on my local copy :) I'll backport the change

File: warpscript/utapi/decodeCheckpoint.mc2 (hunk 1 above)
Comment (tmacro) [reply warpscript/utapi/decodeCheckpoint.mc2:14]: Added tests for encode/decodeEvent
encodeCheckpoint is generalized to encodeRecord in the snapshot PR with tests

**Review by jonathan-gramain — APPROVED** [review]

**Pull request merged**